src/coreclr/interpreter/interpretershared.h

```diff
@@ -17,7 +17,10 @@
 #define INTERP_STACK_SLOT_SIZE 8    // Alignment of each var offset on the interpreter stack
 #define INTERP_STACK_ALIGNMENT 16   // Alignment of interpreter stack at the start of a frame
 
-#define INTERP_INDIRECT_HELPER_TAG 1 // When a helper ftn's address is indirect we tag it with this tag bit
+struct InterpHelperData {
+    int32_t accessType;
+    int32_t addressDataItemIndex;
```

Comment:
If you also agree, I think it would be good to have this be just int32 in size. We could have 2-3 bits for `accessType` and the rest for `addressDataItemIndex`. We would group this helper data in a single slot and avoid increases in instruction sizes.

Reply:
Hm, so we'd bitpack the instruction data, but not the actual pointer in the data item. That sounds reasonable. I'll take a swing at doing it.